### PR TITLE
Correct spelling errors in documentation

### PR DESCRIPTION
**Pull request by rbberger**
This PR starts the process of correcting spelling errors across the documentation and the core source files. It also adds a utility to check for spelling errors in the future.

### CODE REVIEW

**Comment by rbberger** [comment]
ok. Let's try this via GitHub. @jewettaij can you please have a look at the minor changes I made in moltemplate, specifically in this [commit](https://github.com/lammps/lammps/pull/399/commits/dd76ac5010aeb43900f3dd9db9a4a75acf091f59)? It's just some typos. It would be great if you could backport these changes so next time we sync (LAMMPS) they won't get lost. If you want I can do a PR on the moltemplate repo, but it's a bit of an overkill for such a small change.

**Comment by rbberger** [comment]
@jewettaij probably not, I only had errors from A-F so far. And you also have to keep in mind I only find typos inside of the LAMMPS documentation. I only found those in moltemplate by doing a grep while looking for similar mistakes.

**Comment by rbberger** [comment]
Ok, that should be it. @jewettaij there's a few more in this commit (08baaa9)

File: src/USER-MISC/fix_imd.cpp [hunk 1]
```diff
@@ -405,7 +405,7 @@ static int imd_recv_fcoords(void *, int32, float *);
 /** Prepare IMD data packet header */
 static void imd_fill_header(IMDheader *header, IMDType type, int32 length);
 /** Write data to socket */
-static int32 imd_writen(void *s, const char *ptr, int32 n);
+static int32 imd_written(void *s, const char *ptr, int32 n);
```
Comment (akohlmey) [inline src/USER-MISC/fix_imd.cpp]: the changes to this file here are wrong.
the name of the function is supposed to be imd_writen (i.e. write n bytes over IMD connection).

**Pull request merged**